Lib/test/test_zipfile.py

```diff
@@ -1727,6 +1727,17 @@ def test_empty_file_raises_BadZipFile(self):
             fp.write("short file")
         self.assertRaises(zipfile.BadZipFile, zipfile.ZipFile, TESTFN)
 
+    def test_negative_central_directory_offset_raises_BadZipFile(self):
+        # Zip file containing an empty EOCD record
+        buffer = bytearray(b'PK\x05\x06' + b'\0'*18)
+
+        # Set the size of the central directory bytes to become 1,
+        # causing the central directory offset to become negative
+        buffer[12] = 1
+
+        f = io.BytesIO(buffer)
+        self.assertRaises(zipfile.BadZipFile, zipfile.ZipFile, f)
```

Comment:
```suggestion
        for dirsize in 1, 2**32-1:
            buffer[12:16] = struct.pack('<L', dirsize)
            f = io.BytesIO(buffer)
            self.assertRaises(zipfile.BadZipFile, zipfile.ZipFile, f)
```
With your previous code this test could raise OverflowError on 32-bit platform. It is worth to add this test even if your current code is not vulnerable, to prevent a regression.

Reply:
Resolved in f0ee1c7